js/components/utils.js

```diff
@@ -431,34 +431,38 @@ var Utils = {
    * @function arrayDeepMerge
    * @param {Array} target - An array that will receive new items if additional items are passed
    * @param {Array} source - An array containing additional items to merge into target
-   * @param {Object} optionsArgument - optional parameters passed, i.e. arrayMerge function
+   * @param {Object} optionsArgument - optional parameters passed, i.e. arrayMerge, swap, unionBy, clone
    * @returns {Array} new merged array with items from both target and source
    */
   arrayDeepMerge: function(target, source, optionsArgument) {
+    var targetArray = optionsArgument.swap ? source : target;
+    var sourceArray = optionsArgument.swap ? target : source;
     var self = this;
     var destination = [];
-    destination = source.slice();
-    target.forEach(function(targetItem, i) {
+    destination = sourceArray.slice();
```

Comment:
do we need to validate that the sourceArray is actually an array, since anything can be passed in?